initial list of project ideas

orbeckst opened the pull request:
- start work on #4 
- transferred all ideas from Google Doc
- fleshed out
  - Trajectory Clustering
  - SASA (copied from https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-2020-Project-Ideas and edited)
- added a template section
- all other ideas are TODO

orbeckst committed:
initial list of project ideas

- transferred all ideas from Google Doc
- fleshed out
  - Trajectory Clustering
  - SASA (copied from https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-2020-Project-Ideas
    and edited)
- added a template section

IAlibay commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,182 @@
+- Hill Reilley, Pickett angles
+- ...
+
+## Converters for other libraries ##
```
Can we keep converters out of the list of things for now? (as per recent conversations)

orbeckst replied:
sure

IAlibay reviewed (commented):
I think @fiona-naughton was about to send me some doc with some stuff she's been working on for this - do you mind if we just take over this PR?

orbeckst committed:
fix markdown link for UserGuide

orbeckst committed:
removed converters from list

orbeckst commented:
@fiona-naughton @IAlibay  if you already have something, just put up a PR and don't send stuff back and forth behind the scenes, that feels like wasting time. 

Please take over the PR — I really just want to make sure that we are moving forward somehow as I hadn't seen anything happening and we set ourselves a deadline for tomorrow for this list.

fiona-naughton commented:
thanks @orbeckst ; apologies for not just getting my in-work stuff up earlier; I can add it into here now

fiona-naughton committed:
Update projects.md

fiona-naughton committed:
Update README.md

fiona-naughton commented on hackathon/mdakits/README.md:
```diff
@@ -34,7 +34,57 @@ MDAKits paper
+Read our guide on making an MDAKit (https://mdakits.mdanalysis.org/makingakit.html).
+(Additional materials TBA.)
+
+Some other things to keep in mind:
```
I added this here because these were common things that came up when adding blurbs for the various kit ideas - happy to move if it might fit better elsewhere, though.

lilyminium replied:
LGTM.

fiona-naughton committed:
update new file formats

fiona-naughton committed:
Update projects.md

fiona-naughton committed:
Update projects.md

fiona-naughton committed:
Update projects.md

fiona-naughton commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+each cluster can allow for further comparison and visualization.  
+
+### Where to start ###
+- (Independently or in conjunction with the above, create a
```
Unless I've misinterpreted anything, seems like there will be an initial overlap in the project and the above with creating an easier clustering interface. Not sure if we want to leave them both (and potentially have two kits with different ClusterAnalysis, move this one to be an optional extension of the above, or remove the cluster itself here and make it just a Kit for generating densities given a series of frame indices?

orbeckst replied:
You're right, let's do exactly what you suggest.

orbeckst replied:
I rewrote it to "State-based density analysis"

lilyminium commented on hackathon/mdakits/README.md:
```diff
@@ -34,7 +34,57 @@ MDAKits paper
+
+If you’ve ever performed (or want to perform) any analysis/trajectory 
+manipulation with MDAnalysis that is not already directly implemented, 
+chances are that others will, too and could benefit from the creation 
```
```suggestion
chances are that others will too, and could benefit from the creation 
```

lilyminium commented on hackathon/mdakits/README.md:
```diff
@@ -34,7 +34,57 @@ MDAKits paper
+If you already have code for this - awesome! You can jump right into 
+[turning your code into an MDAKit](https://mdakits.mdanalysis.org/makingakit.html).
+
+If you have an idea but don't yet have code (or it's a work-progress) - 
```
```suggestion
If you have an idea but don't yet have code (or it's a work-in-progress) - 
```

lilyminium commented on hackathon/mdakits/README.md:
```diff
@@ -34,7 +34,57 @@ MDAKits paper
+[turning your code into an MDAKit](https://mdakits.mdanalysis.org/makingakit.html).
+
+If you have an idea but don't yet have code (or it's a work-progress) - 
+that's also awesome! Here is your oppurtunity to get things rolling.
```
```suggestion
that's also awesome! Here is your opportunity to get things rolling.
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+  
+* The GROMOS clustering algorithm is widely used in biomolecular
+  simulations [[Daura 1999]](#Daura1999). (See Issue
+  [#2876](https://github.com/MDAnalysis/mdanalysis/issues/2876and has been requested)
```
```suggestion
  [#2876](https://github.com/MDAnalysis/mdanalysis/issues/2876)
```

I think this was leftover from a previous edit?

lilyminium replied:
Leaving up in case it was meant to be something

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+`ClusterAnalysis` class that allows clustering to be performed
+over 3D coordinate of a selected group.)
+- Create a class, making use of the existing [density analysis]( https://docs.mdanalysis.org/stable/documentation_pages/analysis/density.html#module-MDAnalysis.analysis.density)
+that generates densities from a list of indicies of frames
```
```suggestion
that generates densities from a list of indices of frames
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+as appropriate; and look at how existing supported file formats
+are handled.
+- Create a Kit featuring a new reader/writer class, using `ReaderBase`/`WriterBase`,
+for the new file format. If there are existing tool that could
```
```suggestion
for the new file format. If there are existing tools that could
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+*This is one of the most wanted analysis features missing in MDAnalysis.*
+
+It is often necessary to measure volume and surface area of a
+biomolecule or parts of it over a MD trajectory. MDAnalysis is
```
```suggestion
biomolecule or parts of it over an MD trajectory. MDAnalysis is
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+spatial analysis, e.g. for characterizing membranes or interaction 
+surfaces/binding sites. 
+
+Vornoi tessellation and related algorithims (e.g. Delaunay triangulation) 
```
```suggestion
Voronoi tessellation and related algorithms (e.g. Delaunay triangulation) 
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+surfaces/binding sites. 
+
+Vornoi tessellation and related algorithims (e.g. Delaunay triangulation) 
+has been implemented in scipy (https://docs.scipy.org/doc/scipy/reference/spatial.html#delaunay-triangulation-convex-hulls-and-voronoi-diagrams).
```
```suggestion
have been implemented in scipy (https://docs.scipy.org/doc/scipy/reference/spatial.html#delaunay-triangulation-convex-hulls-and-voronoi-diagrams).
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+
+MDAnalysis currently features tools for assessing puckering in 5-membered 
+rings in our [nucleic acid analysis module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/nuclinfo.html), 
+however such capabilities are lacking for 6 (or higher) membered rings.
```
```suggestion
rings in our [nucleic acid analysis module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/nuclinfo.html); 
however, such capabilities are lacking for 6 (or higher) membered rings.
```

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,274 @@
+  simulations [[Daura 1999]](#Daura1999). (See Issue
+  [#2876](https://github.com/MDAnalysis/mdanalysis/issues/2876and has been requested)
+  
+    "To find clusters of structures in a trajectory the RMSD of atom
```
```suggestion
  
  The following excerpt from Daura et al. describes the algorithm:
    "To find clusters of structures in a trajectory the RMSD of atom
```

lilyminium commented on hackathon/mdakits/README.md:
```diff
@@ -34,7 +34,57 @@ MDAKits paper
+tool as a dependency than to make a Kit “from scratch”. Keep in mind how
+easy it would be to install and move data between MDAnalysis and the
+external implementation, and make sure to check the licence
+before including it in your Kit.
```
```suggestion
before including it in your Kit. For now, MDAKits must remain GPLv3+ compatible.
We recognise that licensing compatibility can be quite complicated and nuanced,
so we encourage you to ask any questions to a core developer.
```

lilyminium replied:
I'll leave this open in case we don't want to encourage this or licensing details change, but I think we probably don't want to rely on individual, new developers to understand licenses.

IAlibay replied:
There's no way licensing would change in the span of two weeks - there's 187 contributors we'd have to contact.

IAlibay replied:
You could add here that the Apache 2.0 license should be watched out for.

lilyminium replied:
```suggestion
before including it in your Kit.  For now, MDAKits must remain GPLv3+ compatible.
We recognise that licensing compatibility can be quite complicated and nuanced;
for example, [software licensed under Apache 2.0 can be included in MDAKits with
GPLv3+ licenses, but MDAnalysis code (GPLv3+) cannot be included in Apache
2.0 projects](https://www.apache.org/licenses/GPL-compatibility.html).
Therefore, we encourage you to ask any questions you might have about licenses
to a core developer.
```

lilyminium replied:
Is this similar to what you had in mind, @IAlibay?

lilyminium reviewed (commented):
Generally LGTM! I'll go through and accept all the minor grammar nitpicks, hope that's ok.

lilyminium committed:
Update hackathon/mdakits/README.md

lilyminium committed:
Update hackathon/mdakits/README.md

lilyminium committed:
Update hackathon/mdakits/README.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

lilyminium committed:
Update hackathon/mdakits/projects.md

orbeckst committed:
Update hackathon/mdakits/projects.md

Co-authored-by: Lily Wang <31115101+lilyminium@users.noreply.github.com>

IAlibay commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,275 @@
+MDAnalysis implements some clustering methods within the ENCORE module 
+for Ensemble Similarity Calculations  (namely, Affinity Propogation and 
+DBSCAN/KMeans, via scikit-learn; as described 
+[in the docs here](https://docs.mdanalysis.org/stable/documentation_pages/analysis/encore/clustering.html#clustering-frontend-mdanalysis-analysis-encore-clustering-clusteringmethod).
```
point to mdaencore here instead?

orbeckst replied:
Done in [b52168a](https://github.com/MDAnalysis/UGM2023/pull/14/commits/b52168a1ebf6de8087a2b38fb8069d83726bca21)

IAlibay commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,275 @@
+2. Write a `SASAnalysis` class using the `AnalysisBase` framework (see
+   the [tutorial on writing your own trajectory analysis][UserGuide
+   AnalysisBase].
+3. create test cases (use existing files in MDA and run external
```
```suggestion
3. Create test cases (use existing files in MDA and run external
```

IAlibay commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,275 @@
+   AnalysisBase].
+3. create test cases (use existing files in MDA and run external
+   implementation for reference)
+4. benchmark performance
```
```suggestion
4. Benchmark performance
```

IAlibay commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,275 @@
+MDAnalysis currently features tools for assessing puckering in 5-membered 
+rings in our [nucleic acid analysis module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/nuclinfo.html); 
+however, such capabilities are lacking for 6 (or higher) membered rings.
+
```
Do we need a where to start here?

1. identify rings
2. calculate CP angles
3. calculate HR angles
4. test against known test cases

orbeckst replied:
I have no idea, that's not something I've ever done. I think you originally proposed the project. We could merge "as is" and then you can update later to make it sound the way that you imagined.

IAlibay replied:
I think what I'm pointing out is an inconsistency across the summaries, we have some with objectives, some with where to start,

orbeckst replied:
If someone wants to make it homogenous then that's fine. I think we should use whatever works. I tend to use "Objectives" as a clear "This is what's needed and when you've done it then you have achieved something" but that might not always be the best way to phrase an open-ended idea.

IAlibay reviewed (approved):
Just a couple of comments otherwise, lgtm!

orbeckst committed:
Apply suggestions from code review

Co-authored-by: Irfan Alibay <IAlibay@users.noreply.github.com>

orbeckst commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,275 @@
+the similarity matrix and thus assign different cluster numbers to the
+different frames of the trajectory. 
+
+MDAnalysis implements some clustering methods within the ENCORE module 
+for Ensemble Similarity Calculations  (namely, Affinity Propogation and 
+DBSCAN/KMeans, via scikit-learn; as described 
+[in the docs here](https://docs.mdanalysis.org/stable/documentation_pages/analysis/encore/clustering.html#clustering-frontend-mdanalysis-analysis-encore-clustering-clusteringmethod).
```
```suggestion
The [mdaencore](https://www.mdanalysis.org/mdaencore/) MDAKit for Ensemble Similarity Calculations 
also implements some clustering methods (namely, Affinity Propogation and 
DBSCAN/KMeans, via scikit-learn; as described 
[in the docs here](https://www.mdanalysis.org/mdaencore/autosummary/mdaencore.clustering.ClusteringMethod.html#module-mdaencore.clustering.ClusteringMethod)).
```

orbeckst committed:
rewrote "clustered density analysis"

Following @fiona-naughton 's suggestion, rewrote the "Clustered DensityAnalysis" to "State-based density analysis"

lilyminium committed:
Update hackathon/mdakits/projects.md

Co-authored-by: Oliver Beckstein <orbeckst@gmail.com>

lilyminium commented on hackathon/mdakits/projects.md:
```diff
@@ -0,0 +1,298 @@
+- Consider allowing the calculated secondary structure to be read
+back alongside the trajectory e.g. using the [Auxilliary
+module](https://docs.mdanalysis.org/stable/documentation_pages/auxiliary/init.html),
+and allow for atomselections based on secondary structure.
```
```suggestion
and allow for atom selections based on secondary structure.
```

lilyminium committed:
Update hackathon/mdakits/projects.md

ianmkenney reviewed (approved):
This all seems good to me!

orbeckst commented:
@fiona-naughton do you want to merge it and then we raise specific "fix this" issues?

fiona-naughton commented:
Yup - I'll go ahead and merge now; thanks all!
